### ISSUE TITLE
createUser does not work with a public.users table

### ISSUE THREAD

**Issue by GaryAustin1** [supabase/gotrue#671]
# Bug report

## Describe the bug
A user in Discord (jxyz) was using api.createUser() and encountering:
`  error: {
    message: 'duplicate key value violates unique constraint "users_pkey"',
    status: 500
}`

I suspect this recent change may be the issue: [refactor: remove auth schema prefix](https://github.com/supabase/gotrue/pull/669) 

The user changed the name of the public user table, trigger and function to solve the issue.
I did not discuss details of how the duplicate primary key was being generated, but this was a testing environment. 
 
Edit: Their trigger function used `public.users` and is in the public schema so I don't think it was trying to insert into auth.users.

Edit2:  On an older instance I am NOT able to reproduce this with a public.users table and NO Trigger function.  The Discord user's instance was created in the past 24hours.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:
create a public.users table (normally used to copy user info with a trigger function on auth.users table.
Run below with the service_role key:
```
const email = 'alice@gmail.com';
const password = 'test1234';
const create_user_response = await supabase.auth.api.createUser({ email, password, email_confirm: true });
```

## Expected behavior
Should be able to have a public.users table

**Comment by kangmingtay** [comment]
hey @GaryAustin1, just to clarify, we haven't merged in #669 yet so it shouldn't related to this bug

**Comment by GaryAustin1** [comment]
@kangmingtay Thanks, I can't reproduce the user's issue, so I suspect they actually did have a problem in their trigger function and it got fixed when they renamed the public.user to public.profiles.
